src/regpiler.py

```diff
@@ -0,0 +1,152 @@
+            # I know that re.IGNORECASE is a thing but without it the regex looks more cursed.
+            r'(?P<indentation> +)?(?:(?P<third_const>[Cc][Oo][Nn][Ss][Tt]) +(?=[Cc][Oo][Nn][Ss][Tt] +[Cc][Oo][Nn][Ss][Tt]))?(?P<invalid_mix>^[Cc][Oo][Nn][Ss][Tt] +(?= +[Vv][Aa][Rr] +(?=[Vv][Aa][Rr]|[Cc][Oo][Nn][Ss][Tt])))?(?P<first_const>[Cc][Oo][Nn][Ss][Tt]|[Vv][Aa][Rr]) +(?P<second_const>[Cc][Oo][Nn][Ss][Tt]|[Vv][Aa][Rr]) +(?P<var_name>[^ +\-*/<>=()\[\]!;:.{}]+)(?:<(?P<lifetime>.*)>)? *(?P<assignment_operator>[+-/*]?)= *(?P<value>[^!\n?]+)',
+            line):
+        if match.group("invalid_mix"):
+            raise ValueError("You thought that having const or var three times without having all of them being const "
+                             "was a good idea? Well it isn't so fix it")
+        indentation = match.group("indentation") if match.group("indentation") else ""
+        if indentation and len(indentation) % 3 != 0:
+            raise ValueError("What a strange indentation scheme that you use, this could confuse someone please use the"
+                             "officially recognized 3 space indentation system, thank you, error occurred in\n" + line)
+
+        var_type = 'let' if match.group("first_const").lower() == 'var' else 'const'
+        lifetime = -1
+        if match.group("lifetime") is not None:
+            lifetime_match = match.group("lifetime")
+            if lifetime_match[-1] != 's' and lifetime_match.lower() != 'infinity':
+                futures[int(lifetime_match)] = [f'{lifetime_match}.kill();']
+            elif lifetime_match[-1] == 's':
+                lifetime = int(lifetime_match[:-1])
+            else:
+                lifetime = 'infinity'
+
+        if match.group("third_const"):
+            # TODO implement const const const
+            pass
+
+        value = match.group("value")
+        if match.group("assignment_operator"):
+            value = f'{match.group("var_name")} {match.group("assignment_operator")} {match.group("value")}'
+        return f'{indentation}assign("{match.group("second_const")}", "{value}", {str(var_type == "let").lower()}, {priority}, {lifetime});', futures
+    # single line function, in the case of the multi-line one code is "{"
+    if match := re.match(
+            r'(?= *[functio])((?P<indentation> +)?(?P<function>f?u?n?c?t?i?o?n?) )+(?P<name>.+?) *(?P<parameters>\(.*?\)) +=> +(?P<code>.+)',
+            line,
+            re.IGNORECASE
+    ):
+        func_keyword = match.group("function")
+        func_name = match.group("name")
+        parameters = match.group("parameters")
+        # code syntax should be checked
+        code = match.group("code")
+        line = line.replace(func_keyword, "function").replace("=>", "")
+        return line, futures
+
+    # replace the previous keyword with the function call (not sure if this is right)
+    line = re.sub(r'previous +(?!=[()]*)([^?! ]*)', r'get_var("\1").previous()', line, 1, re.IGNORECASE)
```

Comment:
Not quite right because we need to be able to handle cases of multiple `previous`'s, i.e.:

`const const y = previous previous x`
`previous previous x = 7`